pkg/ds/daemon_set.go

```diff
@@ -268,7 +278,15 @@ func (ds *daemonSet) handleNodeChanges(changes *labels.LabeledChanges) error {
 		ds.logger.NoFields().Infof("Received node change signal")
 		err := ds.removePods()
 		if err != nil {
-			return util.Errorf("Unable to remove pods from intent tree: %v", err)
+			err = util.Errorf("Unable to remove pods from intent tree: %v", err)
+			// We must still try to add pods so that we can label the pods that
+			// potentially need to be scheduled because all updates are taken in
+			// only once
+			subErr := ds.addPods()
+			if subErr != nil {
+				err = util.Errorf("%v, Unable to add pods to intent tree: %v", err, subErr)
+			}
+			return err
```

Comment:
This is because the `removePods()` function will account for any nodes that gets deleted and the `addPods()` will account for any nodes that gets created.